### PR TITLE
Fully developed application

### PR DESCRIPTION
**Pull request by El-Potato-Slayer**
The branch was used to develop a functional blog application with user authentication and authorization

The program:
- Allows users to sign up, sign in, and log out.
- Users who aren't signed in can see the posts but not the authors.
- Users aren't able to edit the posts of other users.
- Users who try to create a post if they're not signed in will be redirected to the sign in page

### CODE REVIEW

File: app/views/layouts/application.html.erb
Comment (lekegitrepo) [inline app/views/layouts/application.html.erb]: - **_[Optional] Change it to `<%= link_to 'Members Only', posts_path, class: "navbar-brand" %>`. It will improve the navigation of the webpage_**

File: config/routes.rb
Comment (lekegitrepo) [inline config/routes.rb]: - **_The application routes uses `resources :posts, only: [:new, :create, :index]` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/posts_routes_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))_**

File: app/controllers/posts_controller.rb
Comment (lekegitrepo) [inline app/controllers/posts_controller.rb]: - **_The `#new` action method should initialize an empty post `@post = Post.new` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/posts_new_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))_**

**Review by lekegitrepo — CHANGES_REQUESTED** [review]
## Status: Changes Required! :recycle:
### Hi @El-Potato-Slayer and @sinansevgi :clap: :clap:
#### Well done :muscle: you did a good job the contents in the view are well formatted.
#### However, you need to fix/implement the following requirements before this project can be approved.

#### PostsController:
- **_`new` method should initialize an empty Post_**

#### Route:
- **_Route definition should be limited only to available and implemented action methods_**

#### Optional:
- **_Add validations to check user input to ensure that the appropriate values are passed and implement Flash messages for unsuccessful actions_**

#### _Reviewed by Temitayo Adeleke_

File: config/routes.rb
Comment (El-Potato-Slayer) [reply config/routes.rb]: I updated the resources, however, I needed to add additional routes since our app has additional features such as show, edit, and delete.

File: README.md
Comment (teekaytech) [inline README.md]: The license link is broken. Kindly fix it.

File: app/views/posts/index.html.erb
Comment (teekaytech) [inline app/views/posts/index.html.erb]: This logic should be moved to a helper file. 
There should be absolutely NO logic in views. This is because views are only for displaying information that became available to it from its corresponding controller not generating it.

File: app/views/posts/show.html.erb
Comment (teekaytech) [inline app/views/posts/show.html.erb]: This is another logic that must be moved to a helper file.

**Review by teekaytech — CHANGES_REQUESTED** [review]
## Status: Changes Required :warning: :vertical_traffic_light: 

### Hello team :wave:

__You've done really well in this project. With the extra features and all, everything is working fine. I should approve this project, if not for the fact that there is one section-specific requirement that has not been satisfied. Please check my comments below and fix them as appropriate.__

**[ REQUIRED CHANGES ]**

- [x] One of the crucial requirements for this module (Rails) is that we should always separate our logic from our views. This rule is not observed in some of your view files (mentioned in the inline comments below). Kindly separate your logic from the view by using the helper file(s). 
**These resources might help**
  - [Where Do I Put My Code?](http://codefol.io/posts/Where-Do-I-Put-My-Code)
  - [The Beginner's Guide to Rails Helpers](https://mixandgo.com/learn/the-beginners-guide-to-rails-helpers)
  - [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/no_logic_in_helpers_requirement.png) from Pathwright

- [x] Your license link is broken. Kindly fix it.

That's it!

* * *
**Please make the required changes and submit another review request when you are done.**

* * *

**Taofeek Olalere
[Twitter](https://twitter.com/ola_lere) | [Github](https://github.com/teekaytech) | @taofeekolalere on Slack**
__Happy Coding :smiley:__

File: app/views/posts/index.html.erb
Comment (El-Potato-Slayer) [reply app/views/posts/index.html.erb]: gg wp

File: app/views/posts/show.html.erb
Comment (El-Potato-Slayer) [reply app/views/posts/show.html.erb]: gg

File: README.md
Comment (El-Potato-Slayer) [reply README.md]: gg

**Review by voscarmv — APPROVED** [review]
# Project status: Approved ✨ 😎👍 💯 ✔️

- Good work moving the logic to the helper methods.
- Good work fixing your license link.

### You may merge this branch and move forward ⛙🛣

### Questions or comments? Reach me on [Twitter](https://twitter.com/voscarmv) or [LinkedIn](https://linkedin.com/in/oscar-mier) 🧑‍💻🗨️🧑‍💻 
# Happy Hacking! 🧠 ⌨️ 🖱️ 🖥️

**Comment by El-Potato-Slayer** [comment]
@voscarmv gg wp 👌

**Pull request merged**
